src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -109,14 +109,20 @@ const UsersFragment = ({
   }, []);
 
   React.useEffect(() => {
+    handleActiveSections();
+  }, [searchText, followedUsers]);
+
+  const handleActiveSections = () => {
+    if (!searchText) return;
+
     if (searchText.length === 0) {
       setActiveSections([]);
     } else if (searchText.length >= 0 && followedUsers.length > 0) {
       setActiveSections([0]);
     } else {
       setActiveSections([]);
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There seems to be some redundancy in your condition checks. If `!searchText` is true, then `searchText.length === 0` will also be true, so there's no need to check both. Also, checking `searchText.length >= 0` is unnecessary because length can't be negative. You can simplify these conditions.

```diff
-  if (!searchText) return;
-
-  if (searchText.length === 0) {
-    setActiveSections([]);
-  } else if (searchText.length >= 0 && followedUsers.length > 0) {
-    setActiveSections([0]);
-  } else {
-    setActiveSections([]);
-  }

+  if (!searchText || searchText.length === 0) {
+    setActiveSections([]);
+  } else if (followedUsers.length > 0) {
+    setActiveSections([0]);
+  } else {
+    setActiveSections([]);
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->